contribute/architecture/k8s-inspired-backend-arch.md

```diff
@@ -0,0 +1,294 @@
+  end
+
+  subgraph "Apps-based API Implementation<br>(apps/... & pkg/registry/apps/...)"
+
+    DA3-->DA1
+    class DA3 disable;
+    DA1-->CUEDefinition
+    class DA1 disable;
+    classDef disable display:none;
+    linkStyle 8 display:none;
+    linkStyle 9 display:none;
+
+    CUEDefinition["CUE Schema Definition<br><i>apps/\*/kinds/\*.cue</i>"] --> |Generates| GeneratedCode["Generated Go/TS Code<br><i>apps/\*/pkg/apis/\*/types.go</i>"]
+    CUEDefinition --> |Defines| Manifest["App Manifest<br><i>apps/\*/kinds/manifest.cue</i>"]
+    AppSDK["Grafana App SDK<br><i>grafana-app-sdk</i>"] --o |Provides| AppSDKTypes["SDK Interfaces/Types<br><i>app.Provider, etc.</i>"]
+    Manifest --> |Configures & Uses| AppRegisterGo["App Registration<br><i>pkg/registry/apps/\*/register.go</i>"]
+    AppSDKTypes --> AppRegisterGo
+    AppRegisterGo --> |Creates| AppProvider["App Provider Instance<br><i>(Implements app.Provider)</i>"]
+    AppRegistryService["App Registry Service<br><i>pkg/registry/apps/apps.go</i>"]
+  end
+
+  AppRegistryService --> |Uses| APIGroupRunner
+  AppProvider --> |Collected by| AppRegistryService
+  APIGroupRunner --> |Creates & Registers<br>AppBuilder with| APIRegistrar
+
+  APIGroupRunner["APIGroupRunner<br><i>pkg/services/apiserver/builder/runner</i>"]
+  APIRegistrar["API Registrar<br><i>pkg/services/apiserver/builder</i>"]
+  Endpoints["API Endpoints<br><i>/apis/\*/v\*/namespaces/\*/resources</i>"]
+  RESTOptionsGetter["RESTOptionsGetter<br><i>Provides storage config</i>"]
+  UnifiedStorage["Unified Storage<br><i>pkg/storage/unified/...</i>"]
+
+  RegistryAPIBuilder --> |Registers builder with| APIRegistrar
+
+  APIRegistrar --> |Builds Server Config leading to| Endpoints
+  APIRegistrar --> |Uses| RESTOptionsGetter
+
+
+  RESTOptionsGetter --> |Connects to| UnifiedStorage
+
+  classDef registry fill:#f5f5f5,stroke:#333,stroke-width:1px;
+  classDef appsStyle fill:#e6f7ff,stroke:#0066cc,stroke-width:1px;
+  classDef shared fill:#f0f0f0,stroke:#666,stroke-width:1px;
+
+  class RegisterAPI,RegistryAPIBuilder,APIGroups,Resources,HandlerFunc,APISchemaRegister,SchemeBuilder,TypeDefs registry;
+  class CUEDefinition,GeneratedCode,Manifest,AppProvider,AppSDK,AppSDKTypes,AppRegistryService,AppRegisterGo appsStyle;
+  class APIRegistrar,Endpoints,UnifiedStorage,BackendDB,RESTOptionsGetter,APIGroupRunner shared;
+```
+
+
+## 4. Migration Path: From Legacy APIs to Resource APIs
+
+**Coexistence During Migration:**
+
+Both API styles will coexist for a while. How they interact depends on the specific resource and the stage of migration. The migration generally follows these steps per resource type:
+
+1. **Define schema:** Define the resource using CUE (Apps Approach). Introduce an Alpha version (`v1alpha1`).
+2. **Implement handlers:** Create Resource API handlers (CRUD) interacting with legacy storage. The initial goal is to mimic the behavior of the legacy API.
```

Comment:
The initial goal is to expose legacy resources via the new Resource API?

Reply:
right - thats the general strategy and thats what we have done for playlists / dashboards / folders / querier etc.

Reply:
ah I think I understand - by `The initial goal is to mimic the behavior of the legacy API.` I wanted to say "do not remodel everything for v1alpha1 - mostly copy existing api structure but define it using cue". does that make sense?

Reply:
rephrased